src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
Better to have readable naming for variables. `ch` is not good, you can call it as `child`. Same for variables `p`, `m`, `w`, `p1`

Reply:
If requirements for the first function `calculateMenAverageAge` are shared with this too, you should avoid `forEach` method. Try to rewrite it using a different array method.